superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -148,6 +169,39 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
     sliceParams.new_dashboard_name = this.state.newDashboardName;
     const { url_params, ...formData } = this.props.form_data || {};
 
+    if (this.props.datasource?.type === 'query') {
+      const { schema, sql, database } = this.props.datasource;
+      const { templateParams } = this.props.datasource;
+      const selectedColumns =
+        this.props.datasource?.results?.selected_columns || [];
+      SupersetClient.post({
+        endpoint: '/superset/sqllab_viz/',
+        postPayload: {
+          data: {
+            schema,
+            sql,
+            dbId: database?.id,
+            templateParams,
+            datasourceName: this.state.datasetName,
+            columns: selectedColumns,
+          },
+        },
+      })
+        .then(({ json }) => json)
+        .then((data: { table_id: number }) => {
+          exploreChart({
+            datasource: `${data.table_id}__table`,
+            metrics: [],
+            groupby: [],
+            time_range: 'No filter',
+            viz_type: 'table',
+            all_columns: selectedColumns.map((c: { name: string }) => c.name),
```

Comment:
do we need to alter `columns` for this to work, can we just send the full json objects?